wordpress-mobile/WordPress-Android
# RI: Choose interests - introduce failed state

Master Issue: #12027

This PR introduces failed state into the Choose Interests screen.
Currently just reused site creation fullscreen error layout to display error message, retry button. 

### To test
- Enable airplane mode on the device
- Launch app and go to Reader tab
- Notice that no network error message is shown along with a Retry button
- Disable airplane mode on the device
- Click Retry
- Notice that progress bar is shown and interests chips are displayed after a while

#### Note
- This PR also refactors code to remove interests related fields from the parent `UiState` and include them only in the `ContentLoadSuccessUiState`.
Ref: https://github.com/wordpress-mobile/WordPress-Android/pull/12113#discussion_r440149546

- As the tags repository currently does not return any errors, so just handling network error.  

- I can see a "No Connection" snackbar when there's no connection. I'm not sure if we'd want to reuse it instead of full screen error layout and how we can retry n/w request in this case (considering we'll also handle API errors later).

PR submission checklist:
